libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -267,12 +267,16 @@ void ContractCompiler::appendFunctionSelector(ContractDefinition const& _contrac
 	m_context << notFound;
 	if (fallback)
 	{
+		m_context.setStackOffset(0);
 		if (!fallback->isPayable())
 			appendCallValueCheck();
 
 		eth::AssemblyItem returnTag = m_context.pushNewTag();
 		fallback->accept(*this);
 		m_context << returnTag;
+		m_context.adjustStackOffset(
+			CompilerUtils(m_context).sizeOnStack(FunctionType(*fallback).returnParameterTypes()) - 1
```

Comment:
The `-1` is the return tag, yes. I will add an assert that the fallback has neither arguments not return parameters and then change  this to just `-1`.